packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -789,55 +797,43 @@ export class StatsAnalyzer extends EventsScope {
 
     LoggerProxy.logger.trace('StatsAnalyzer:index#getStatsAndParse --> Collecting Stats');
 
-    return this.mediaConnection.getTransceiverStats().then((transceiverStats) => {
-      transceiverStats.video.receivers.forEach((receiver, i) =>
-        this.filterAndParseGetStatsResults(receiver, `video-recv-${i}`, false)
-      );
-      transceiverStats.audio.receivers.forEach((receiver, i) =>
-        this.filterAndParseGetStatsResults(receiver, `audio-recv-${i}`, false)
-      );
-      transceiverStats.screenShareVideo.receivers.forEach((receiver, i) =>
-        this.filterAndParseGetStatsResults(receiver, `video-share-recv-${i}`, false)
-      );
-      transceiverStats.screenShareAudio.receivers.forEach((receiver, i) =>
-        this.filterAndParseGetStatsResults(receiver, `audio-share-recv-${i}`, false)
-      );
-
-      transceiverStats.video.senders.forEach((sender, i) => {
-        if (i > 0) {
-          throw new Error('Stats Analyzer does not support multiple senders.');
-        }
-        this.filterAndParseGetStatsResults(sender, 'video-send', true);
-      });
-      transceiverStats.audio.senders.forEach((sender, i) => {
-        if (i > 0) {
-          throw new Error('Stats Analyzer does not support multiple senders.');
-        }
-        this.filterAndParseGetStatsResults(sender, 'audio-send', true);
-      });
-      transceiverStats.screenShareVideo.senders.forEach((sender, i) => {
-        if (i > 0) {
-          throw new Error('Stats Analyzer does not support multiple senders.');
-        }
-        this.filterAndParseGetStatsResults(sender, 'video-share-send', true);
-      });
-      transceiverStats.screenShareAudio.senders.forEach((sender, i) => {
-        if (i > 0) {
-          throw new Error('Stats Analyzer does not support multiple senders.');
-        }
-        this.filterAndParseGetStatsResults(sender, 'audio-share-send', true);
-      });
+    return this.mediaConnection
+      .getTransceiverStats()
+      .then(({video, audio, screenShareVideo, screenShareAudio}) => {
+        const entries = [
+          {type: 'video-recv-', receivers: video.receivers, isSender: false},
+          {type: 'audio-recv-', receivers: audio.receivers, isSender: false},
+          {type: 'video-share-recv-', receivers: screenShareVideo.receivers, isSender: false},
+          {type: 'audio-share-recv-', receivers: screenShareAudio.receivers, isSender: false},
+          {type: 'video-send', receivers: video.senders, isSender: true},
+          {type: 'audio-send', receivers: audio.senders, isSender: true},
+          {type: 'video-share-send', receivers: screenShareVideo.senders, isSender: true},
+          {type: 'audio-share-send', receivers: screenShareAudio.senders, isSender: true},
+        ];
+        const tasks = entries.map(({receivers, type, isSender}, idx) => {
+          return this.taskQueue.addTask(() => {
```

Comment:
I'm not sure if we need this solution at all, but if we do, then the call to `getTransceiverStats()` is the one that actually does the call to `getStats()` and iterates all the reports (this is done inside WCME lib), so I think it should be the one on the taskQueue, not the calls to `filterAndParseGetStatsResults()`.